### ISSUE TITLE
Clef changes at system break

### ISSUE THREAD

**Issue by craigsapp** [rism-ch/verovio#236]
Clef changes at system breaks are incorrect in the following verovio rendering:

![screen shot 2016-07-11 at 11 15 11 pm](https://cloud.githubusercontent.com/assets/3487289/16756942/848e350c-47bd-11e6-95ff-10e71d1e22c5.png)

There should be a small bass clef at the end of staff 1 in the first system.  Then at the start of staff 1 on the second system, there should be a large bass clef.

Also, there is a strange blank gap after the repeat sign at the end of the first measure...

MEI data for test:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://relaxng.org/ns/structure/1.0"?>
<?xml-model href="http://music-encoding.org/schema/3.0.0/mei-all.rng" type="application/xml" schematypens="http://purl.oclc.org/dsdl/schematron"?>
<mei xmlns="http://www.music-encoding.org/ns/mei" meiversion="3.0.0">
    <meiHead>
        <fileDesc>
            <titleStmt>
                <title>Piano Sonata no. 14 in C-sharp minor ("Moonlight"), mvmt. 3</title>
                <respStmt>
                    <persName role="Composer">Beethoven, Ludwig van</persName>
                    <persName role="Dedicatee">Gräfin Giulietta Guicciardi</persName>
                </respStmt>
            </titleStmt>
            <pubStmt>
                <date>2016-07-11 23:14:21</date>
            </pubStmt>
        </fileDesc>
        <encodingDesc>
            <projectDesc>
                <p>Transcoded from Humdrum with Verovio version 0.9.12-dev-48e949d-dirty</p>
                <p>Encoded by: Craig Stuart Sapp</p>
            </projectDesc>
        </encodingDesc>
    </meiHead>
    <music>
        <body>
            <mdiv>
                <score>
                    <scoreDef xml:id="scoredef-000000108570039">
                        <staffGrp xml:id="m-000000192794127" symbol="brace">
                            <staffDef xml:id="staffdef-000000167567576" clef.shape="F" clef.line="4" key.sig="4s" meter.count="4" meter.unit="4" n="1" lines="5" />
                            <staffDef xml:id="staffdef-000000098196836" clef.shape="F" clef.line="4" key.sig="4s" meter.count="4" meter.unit="4" n="2" lines="5" />
                        </staffGrp>
                    </scoreDef>
                    <section>
                        <measure xml:id="measure-000000053055058" n="1" right="rptstart">
                            <staff xml:id="staff-000000016355660" n="1">
                                <!-- kern: =1-  16r  16GG#/LL  16C#/  16E/JJ  16G#/LL  16C#/  16E/  16G#/JJ  16c#/LL  16E/  16G#/  16c#/JJ  *clefG2  16e/LL  16G#/  16c#/  16e/JJ  =2!|: -->
                                <layer xml:id="layer-000000116587149" n="1">
                                    <rest xml:id="rest-000000116142123" dur="16" />
                                    <beam xml:id="beam-000000152786267">
                                        <note xml:id="note-000000132602835" dur="16" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000195059000" dur="16" oct="3" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000008092616" dur="16" oct="3" pname="e" accid.ges="n" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000076883937">
                                        <note xml:id="note-000000047428843" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000205387941" dur="16" oct="3" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000089916922" dur="16" oct="3" pname="e" accid.ges="n" stem.dir="up" />
                                        <note xml:id="note-000000049465660" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000078429547">
                                        <note xml:id="note-000000039933900" dur="16" oct="4" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000080424335" dur="16" oct="3" pname="e" accid.ges="n" stem.dir="up" />
                                        <note xml:id="note-000000065595965" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000168032792" dur="16" oct="4" pname="c" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <clef xml:id="clef-000000186139339" shape="G" line="2" />
                                    <beam xml:id="beam-000000204441988">
                                        <note xml:id="note-000000082658796" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="up" />
                                        <note xml:id="note-000000039224892" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000190038663" dur="16" oct="4" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000027395984" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="up" />
                                    </beam>
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000011675846" n="2">
                                <!-- kern: =1-  8CC#''\L  8GG#''\  8CC#''\  8GG#''\J  8CC#''\L  8GG#''\  8CC#''\  8GG#''\J  =2!|: -->
                                <layer xml:id="layer-000000023812532" n="1">
                                    <beam xml:id="beam-000000141031972">
                                        <note xml:id="note-000000146653890" dur="8" oct="2" pname="c" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000144961013" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000039559734" dur="8" oct="2" pname="c" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000019518949" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                    </beam>
                                    <beam xml:id="beam-000000134233030">
                                        <note xml:id="note-000000122969078" dur="8" oct="2" pname="c" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000003042157" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000019428623" dur="8" oct="2" pname="c" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000119355778" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                    </beam>
                                </layer>
                            </staff>
                            <tempo xml:id="tempo-000000170688431" staff="1">Presto agitato</tempo>
                            <dynam xml:id="dynam-000000212072395" staff="1" tstamp="1.250000">p</dynam>
                        </measure>
                        <measure xml:id="measure-000000048092906" n="2">
                            <staff xml:id="staff-000000047287292" n="1">
                                <!-- kern: =2!|:  *>A  16g#/LL  16c#/  16e/  16g#/JJ  16cc#/LL  16e/  16g#/  16cc#/JJ  16ee\LL  16g#\  16cc#\  16ee\JJ  8g#''\ 8cc#''\ 8ee''\ 8gg#''\L  8g#''\ 8cc#''\ 8ee''\ 8gg#''\J  =3 -->
                                <layer xml:id="layer-000000021719260" n="1">
                                    <beam xml:id="beam-000000178136236">
                                        <note xml:id="note-000000128779614" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000164968177" dur="16" oct="4" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000004025962" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="up" />
                                        <note xml:id="note-000000018611814" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000135154021">
                                        <note xml:id="note-000000140179195" dur="16" oct="5" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000202183051" dur="16" oct="4" pname="e" accid.ges="n" stem.dir="up" />
                                        <note xml:id="note-000000127173593" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000016105372" dur="16" oct="5" pname="c" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000100057766">
                                        <note xml:id="note-000000191180922" dur="16" oct="5" pname="e" accid.ges="n" stem.dir="down" />
                                        <note xml:id="note-000000112730135" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000145312284" dur="16" oct="5" pname="c" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000145160542" dur="16" oct="5" pname="e" accid.ges="n" stem.dir="down" />
                                    </beam>
                                    <beam xml:id="beam-000000171813124">
                                        <chord xml:id="chord-000000156676757" dur="8" stem.dir="down">
                                            <note xml:id="note-000000021813670" oct="4" pname="g" accid.ges="s" />
                                            <note xml:id="note-000000046899869" oct="5" pname="c" accid.ges="s" />
                                            <note xml:id="note-000000119614960" oct="5" pname="e" accid.ges="n" />
                                            <note xml:id="note-000000109195805" oct="5" pname="g" accid.ges="s" />
                                        </chord>
                                        <chord xml:id="chord-000000014376631" dur="8" stem.dir="down">
                                            <note xml:id="note-000000036137013" oct="4" pname="g" accid.ges="s" />
                                            <note xml:id="note-000000046413884" oct="5" pname="c" accid.ges="s" />
                                            <note xml:id="note-000000112092839" oct="5" pname="e" accid.ges="n" />
                                            <note xml:id="note-000000171701689" oct="5" pname="g" accid.ges="s" />
                                        </chord>
                                    </beam>
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000188568934" n="2">
                                <!-- kern: =2!|:  *>A  8CC#''/L  8GG#''/  8CC#''/  8GG#''/J  8CC#''/L  8GG#''/  8CC#''/ 8C#''/  8GG#''/J  =3 -->
                                <layer xml:id="layer-000000001770587" n="1">
                                    <beam xml:id="beam-000000122996506">
                                        <note xml:id="note-000000034529504" dur="8" oct="2" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000087304073" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000158735684" dur="8" oct="2" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000051708000" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000184482508">
                                        <note xml:id="note-000000060622417" dur="8" oct="2" pname="c" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000114727105" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                        <chord xml:id="chord-000000207635458" dur="8" stem.dir="up">
                                            <note xml:id="note-000000068222953" oct="2" pname="c" accid.ges="s" />
                                            <note xml:id="note-000000081665383" oct="3" pname="c" accid.ges="s" />
                                        </chord>
                                        <note xml:id="note-000000093298325" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-000000187144323" n="3">
                            <staff xml:id="staff-000000032979775" n="1">
                                <!-- kern: =3  *clefF4  16r  16GG#/LL  16BB#/  16D#/JJ  16G#/LL  16BB#/  16D#/  16G#/JJ  16B#/LL  16D#/  16G#/  16B#/JJ  16d#/LL  16G#/  16B#/  16d#/JJ  =4 -->
                                <layer xml:id="layer-000000122319144" n="1">
                                    <clef xml:id="clef-000000031771380" shape="F" line="4" />
                                    <rest xml:id="rest-000000117157419" dur="16" />
                                    <beam xml:id="beam-000000036990240">
                                        <note xml:id="note-000000213209683" dur="16" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000123942242" dur="16" accid="s" oct="2" pname="b" stem.dir="up" />
                                        <note xml:id="note-000000038130426" dur="16" oct="3" pname="d" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000048962962">
                                        <note xml:id="note-000000004782249" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000059373906" dur="16" oct="2" pname="b" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000176345830" dur="16" oct="3" pname="d" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000102183585" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000056845631">
                                        <note xml:id="note-000000203797392" dur="16" accid="s" oct="3" pname="b" stem.dir="up" />
                                        <note xml:id="note-000000201112189" dur="16" oct="3" pname="d" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000168053551" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000105552568" dur="16" oct="3" pname="b" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000200521315">
                                        <note xml:id="note-000000115660690" dur="16" oct="4" pname="d" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000007031330" dur="16" oct="3" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000063971128" dur="16" oct="3" pname="b" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000132443011" dur="16" oct="4" pname="d" accid.ges="s" stem.dir="up" />
                                    </beam>
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000025549134" n="2">
                                <!-- kern: =3  8BBB#''\L  8GG#''\  8BBB#''\  8GG#''\J  8BBB#''\L  8GG#''\  8BBB#''\  8GG#''\J  =4 -->
                                <layer xml:id="layer-000000102890803" n="1">
                                    <beam xml:id="beam-000000131903540">
                                        <note xml:id="note-000000055441427" dur="8" accid="s" oct="1" pname="b" stem.dir="down" />
                                        <note xml:id="note-000000010917559" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000096314020" dur="8" oct="1" pname="b" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000191319480" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                    </beam>
                                    <beam xml:id="beam-000000079240749">
                                        <note xml:id="note-000000144658938" dur="8" oct="1" pname="b" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000116539239" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000169910531" dur="8" oct="1" pname="b" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000177302546" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="down" />
                                    </beam>
                                </layer>
                            </staff>
                        </measure>
                        <measure xml:id="measure-000000075593809" n="4">
                            <staff xml:id="staff-000000127410721" n="1">
                                <!-- kern: =4  *clefG2  16g#/LL  16B#/  16d#/  16g#/JJ  16b#/LL  16d#/  16g#/  16b#/JJ  16dd#\LL  16g#\  16b#\  16dd#\JJ  8g#''\ 8b#''\ 8dd#''\ 8gg#''\L  8g#''\ 8b#''\ 8dd#''\ 8gg#''\J  = -->
                                <layer xml:id="layer-000000140719016" n="1">
                                    <clef xml:id="clef-000000040776597" shape="G" line="2" />
                                    <beam xml:id="beam-000000070237382">
                                        <note xml:id="note-000000007930283" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000140280267" dur="16" accid="s" oct="3" pname="b" stem.dir="up" />
                                        <note xml:id="note-000000182899792" dur="16" oct="4" pname="d" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000088718551" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000094790544">
                                        <note xml:id="note-000000141318789" dur="16" accid="s" oct="4" pname="b" stem.dir="up" />
                                        <note xml:id="note-000000027984905" dur="16" oct="4" pname="d" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000043394768" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000050424296" dur="16" oct="4" pname="b" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000084110892">
                                        <note xml:id="note-000000178027069" dur="16" oct="5" pname="d" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000011988360" dur="16" oct="4" pname="g" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000054400431" dur="16" oct="4" pname="b" accid.ges="s" stem.dir="down" />
                                        <note xml:id="note-000000124262011" dur="16" oct="5" pname="d" accid.ges="s" stem.dir="down" />
                                    </beam>
                                    <beam xml:id="beam-000000043787295">
                                        <chord xml:id="chord-000000205183048" dur="8" stem.dir="down">
                                            <note xml:id="note-000000082254106" oct="4" pname="g" accid.ges="s" />
                                            <note xml:id="note-000000109539329" oct="4" pname="b" accid.ges="s" />
                                            <note xml:id="note-000000204528698" oct="5" pname="d" accid.ges="s" />
                                            <note xml:id="note-000000036755213" oct="5" pname="g" accid.ges="s" />
                                        </chord>
                                        <chord xml:id="chord-000000128581459" dur="8" stem.dir="down">
                                            <note xml:id="note-000000055794159" oct="4" pname="g" accid.ges="s" />
                                            <note xml:id="note-000000141081797" oct="4" pname="b" accid.ges="s" />
                                            <note xml:id="note-000000125079291" oct="5" pname="d" accid.ges="s" />
                                            <note xml:id="note-000000035905150" oct="5" pname="g" accid.ges="s" />
                                        </chord>
                                    </beam>
                                </layer>
                            </staff>
                            <staff xml:id="staff-000000136048465" n="2">
                                <!-- kern: =4  8BBB#''/L  8GG#''/  8BBB#''/  8GG#''/J  8BBB#''/L  8GG#''/  8BBB#''/ 8BB#''/  8GG#''/J  = -->
                                <layer xml:id="layer-000000014952923" n="1">
                                    <beam xml:id="beam-000000058201926">
                                        <note xml:id="note-000000020984199" dur="8" accid="s" oct="1" pname="b" stem.dir="up" />
                                        <note xml:id="note-000000064631201" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000059369915" dur="8" oct="1" pname="b" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000109272454" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                    <beam xml:id="beam-000000014126186">
                                        <note xml:id="note-000000121871831" dur="8" oct="1" pname="b" accid.ges="s" stem.dir="up" />
                                        <note xml:id="note-000000029966150" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                        <chord xml:id="chord-000000056176232" dur="8" stem.dir="up">
                                            <note xml:id="note-000000120120002" oct="1" pname="b" accid.ges="s" />
                                            <note xml:id="note-000000007510514" accid="s" oct="2" pname="b" />
                                        </chord>
                                        <note xml:id="note-000000171933845" dur="8" oct="2" pname="g" accid.ges="s" stem.dir="up" />
                                    </beam>
                                </layer>
                            </staff>
                        </measure>
                    </section>
                </score>
            </mdiv>
        </body>
    </music>
</mei>
```

**Comment by lpugin** [comment]
The clef change should appear at the end of measure 2 and not in measure 3. I guess this would also be the case without system break since usually a clef change in written in the preceding measure.

**Comment by donbyrd** [comment]
Gould (p. 8) makes it clear that it should be "before the barline", i.e., at the end of m. 2, regardless of the system break. This makes sense because if it was at the beginning of a measure it'd almost certainly require horizontal space of its own,  while if it's at the end of the previous measure there's a good chance it can share space with something happening on other staves, for example short notes.

**Comment by lpugin** [comment]
This needs to be fixed in the encoding.

**Comment by pe-ro** [comment]
A naive question -- 

Is it possible for Verovio to supply a cautionary clef when a system break (encoded or necessitated by a change in layout) precedes a measure in which the clef changes?  For example, at the end of m. 1 when m. 2

- is preceded by a `<staffDef>` that declares a new clef --
```
<section xmlns="http://www.music-encoding.org/ns/mei">
  <staffDef n="1" clef.line="2" clef.shape="G" lines="5"/>
  <measure n="1">
    <!-- notes, rests, etc. -->
  </measure>
  <staffDef n="1" clef.line="4" clef.shape="F"/>
  <measure n="2">
    <!-- notes, rests, etc. -->
  </measure>
</section>
```
- or has a `<staffDef>` or `<clef>` descendant which re-declares the clef --
```
<section xmlns="http://www.music-encoding.org/ns/mei">
  <staffDef n="1" clef.line="2" clef.shape="G" lines="5"/>
  <measure n="1">
    <!-- notes, rests, etc. -->
  </measure>
  <measure n="2">
    <staffDef n="1" clef.line="4" clef.shape="F"/>
    <staff n="1">
      <layer>
        <!-- notes, rests, etc. -->
      </layer>
    </staff>
  </measure>
</section>
```
```
<section xmlns="http://www.music-encoding.org/ns/mei">
  <staffDef n="1" clef.line="2" clef.shape="G" lines="5"/>
  <measure n="1">
    <!-- notes, rests, etc. -->
  </measure>
  <measure n="2">
    <staff n="1">
      <staffDef n="1" clef.line="4" clef.shape="F"/>
      <layer>
        <!-- notes, rests, etc. -->
      </layer>
    </staff>
  </measure>
</section>
```
```
<section xmlns="http://www.music-encoding.org/ns/mei">
  <staffDef n="1" clef.line="2" clef.shape="G" lines="5"/>
  <measure n="1">
    <!-- notes, rests, etc. -->
  </measure>
  <measure n="2">
    <staff n="1">
      <layer>
        <clef line="4" shape="F"/>
        <!-- notes, rests, etc. -->
      </layer>
    </staff>
  </measure>
</section>
```
?

**Comment by craigsapp** [comment]
> This needs to be fixed in the encoding.

Yes, you are right: it is an encoding error.   

With the fix:

![screen shot 2016-07-14 at 4 49 49 pm](https://cloud.githubusercontent.com/assets/3487289/16859354/047647ce-49e3-11e6-9d82-ed9424c6223e.png)

Although I now have a substitute issue to report :-)  The clef changes have too large of a gap to the barline.  The space should be about one staff line space or two diatonic steps, perhaps half of that in tight situations.  Also, the gaps in this case should not be affected by justification stretching (as you can see when comparing the bass clef change on the two lines).

 Here is an example of a clef change in SCORE:

![screen shot 2016-07-14 at 5 02 04 pm](https://cloud.githubusercontent.com/assets/3487289/16859561/b54336d8-49e4-11e6-9e90-23b1f94e64be.png)


Any idea about the sub-issue with the gap in the barlines?:

![screen shot 2016-07-14 at 4 51 58 pm](https://cloud.githubusercontent.com/assets/3487289/16859397/51ac0fb0-49e3-11e6-824f-a7d93245cd23.png)

If I move the rptbegin to the @left of the next measure it looks even more funny:

![screen shot 2016-07-14 at 4 46 59 pm](https://cloud.githubusercontent.com/assets/3487289/16859402/6563093c-49e3-11e6-84c3-84e00f4f6ff9.png)


The original does cross-staff voicing, which I will probably work on matching:

![screen shot 2016-07-14 at 4 40 00 pm](https://cloud.githubusercontent.com/assets/3487289/16859330/eeb8acf6-49e2-11e6-968b-a5e0a29657a9.png)

SharpEye cannot do that, so I did the clef change method on the top staff, but probably did not view the notation after the edit.

> This makes sense because if it was at the beginning of a measure it'd almost certainly require horizontal space of its own, while if it's at the end of the previous measure there's a good chance it can share space with something happening on other staves, for example short notes.

This is true, but the more important reason is that you do not want to surprise the musician with a clef change too late. It is wiser to alert them to the change in the next measure before the system break.

**Comment by lpugin** [comment]
@pe-ro : what is currently being implemented is indeed to show cautionary score definition information at the end of a system when necessary. This will includes clefs, but most importantly time and key signatures.

However: it seems that the case reported by @craigsapp is a different issue, namely where to put the ```clef```. As I said, this case is not directly related to the system break. With no system break there, the clef would still quite likely be placed at the end of the previous measure. Verovio expects these to be encoded as ```clef``` elements in the ```layer```, in the same way as when they occur anywhere  else in the measure. I do not see any reason to encode some as ```clef``` elements and some as ```staffDef```, furthermore after the measure in which they occur.

In other words, the idea is to make the distinction between clefs changes that occur in the flow and full score definition changes. Only these will yield automatic cautionary rendering, but after the bar line of the last measure. As a matter of fact they most of the time occur at all voices and very often do not include a clef change. They will quite likely be encoded with a ```scoreDef``` and not with just ```staffDef``` elements. This somehow relates to https://github.com/rism-ch/verovio/issues/238#issuecomment-232648504

Does this make sense?

@craigsapp : the gap issue was fixed, wasn't it?
